src/management-system-v2/lib/data/db/space-engines.ts

```diff
@@ -0,0 +1,93 @@
+import Ability, { UnauthorizedError } from '@/lib/ability/abilityHelper';
+import { toCaslResource } from '@/lib/ability/caslAbility';
+import db from '@/lib/data/db';
+import { SpaceEngineInput, SpaceEngineInputSchema } from '@/lib/space-engine-schema';
+
+export async function getSpaceEngines(environmentId?: string, ability?: Ability) {
+  const engines = await db.engine.findMany({
+    where: { environmentId: environmentId },
+  });
+
+  return ability ? ability.filter('view', 'Machine', engines) : engines;
+}
+
+export async function getSpaceEngineById(
+  environmentId: string,
+  engineId: string,
+  ability?: Ability,
+) {
+  const engine = await db.engine.findFirst({
```

Comment:
This should be `findUnique()`, right? Or can the engineId be a duplicate?

Reply:
No, it should be unique, the engineId is just a primary key for the table, it has nothing to do with the engine's id inside PROCEED. I'll change it in an upcoming PR